GoRogue/MapGeneration/Region.cs

```diff
@@ -1,355 +1,478 @@
+            _southBoundary = new Area(Lines.Get(SouthWestCorner, SouthEastCorner));
+            _eastBoundary = new Area(Lines.Get(SouthEastCorner, NorthEastCorner));
+            _northBoundary = new Area(Lines.Get(NorthEastCorner, NorthWestCorner));
+
+            _outerPoints = new MultiArea()
             {
-                while (_outerPoints.Contains(c))
-                    _outerPoints.Remove(c);
-                while (_innerPoints.Contains(c))
-                    _innerPoints.Remove(c);
-            }
+                _westBoundary, _northBoundary, _eastBoundary, _southBoundary
+            };
+
+            _innerPoints = InnerFromOuterPoints(OuterPoints);
         }
 
+
+        /// <summary>
+        /// Gets the points that are inside of the provided points
+        /// </summary>
+        /// <param name="outer">An IEnumerable of Points that form a closed region of any shape and size</param>
+        /// <returns>All points contained within the outer region</returns>
+        public static Area InnerFromOuterPoints(IEnumerable<Point> outer)
+            => InnerFromOuterPoints(new Area(outer));
+
         /// <summary>
-        /// Removes (from this region) the inner points shared in common
-        /// with the imposing region
+        /// Gets the points contains within a hollow area
         /// </summary>
-        /// <param name="imposing">the region to check for common outer points</param>
-        public void RemoveOverlappingInnerPoints(Region imposing)
+        /// <param name="outer">The outer area</param>
+        /// <returns></returns>
+        public static Area InnerFromOuterPoints(Area outer)
         {
-            foreach (Point c in imposing._innerPoints)
+            var outerList = outer.OrderBy(x => x.X).ToList();
+
+            if(outerList.Count == 0)
+                return new Area();
+
+            Area points = new Area();
+
+            for (int i = outerList[0].X + 1; i < outerList[^1].X; i++)
             {
-                while (_outerPoints.Contains(c))
-                    _outerPoints.Remove(c);
-                while (_innerPoints.Contains(c))
-                    _innerPoints.Remove(c);
+                List<Point> row = outerList.Where(point => point.X == i).OrderBy(point => point.Y).ToList();
+                if(row.Count > 0)
+                {
+                    for (int j = row[0].Y; j <= row[^1].Y; j++)
+                    {
+                        points.Add(new Point(i, j));
+                    }
+                }
             }
+
+            return points;
         }
+
+        /// <summary>
+        /// Returns a new Region from a parameter set similar to GoRogue.Rectangle
+        /// </summary>
+        /// <param name="origin"></param>
+        /// <param name="width"></param>
+        /// <param name="height"></param>
+        /// <returns>A new region in the shape of a rectangle</returns>
+        public static Region Rectangle(Point origin, int width, int height)
```

Comment:
Rectangle already has a constructor very similar to this; `public Rectangle(int x, int y, int width, int height)`; so if a user wants to do this, would it be better to just say, let them create a rectangle either exactly as you do in the forwarder, or simply `new Rectangle(origin.X, origin.Y, width, height)`?

Minor nitpick, but it feels odd to provide an overload for this particular parameter set for creating a rectangle, but not any of the other ones; particularly when this function has a different name `Rectangle` compared to `FromRectangle`.  There are already a lot of constructors, implicit tuple conversions, etc. Built into `Rectangle`, so I feel like it may be less confusing if the API just depends on those wherever it can.